src/hera_display_games/random_walk/__init__.py

```diff
@@ -2,31 +2,69 @@
 # -*- coding: utf-8 -*-
 """A simple simulation of a random walk"""
 import click
-from hera_display_games.core import mechanics
+from hera_display_games.core import mechanics, sprites, keymapper, map_dict
+import asyncio
 import time
 import random
 
 
+async def update_board(board, speed=10.0):
+    directions = ["r", "l", "dl", "dr", "ur", "ul"]
+    while True:
+        for sprite in board.sprites:
+            board.move_sprite(sprite, random.choice(directions))
+
+        board.draw()
+        await asyncio.sleep(1.0 / speed)
+
+
+# event loop and other code adapted from
+# https://github.com/AlexElvers/pygame-with-asyncio
+def pygame_event_loop(loop, event_queue):
+    import pygame
+
+    while True:
+        event = pygame.event.wait()
+        asyncio.run_coroutine_threadsafe(event_queue.put(event), loop=loop)
```

Comment:
This actually isn't necessary here if we aren't using the keyboard interface, the pygame event queue, or pygame event capturing. This function is only used to capture the pygame listener events (which happen in a blocked thread) and place them into an ansycio event_queue which can be put in the background.

Reply:
Right. I'll try fix that up.